### ISSUE TITLE
mutli person keypoint detection

### ISSUE THREAD

**Issue by soans1994** [robertklee/COCO-Human-Pose#40]
Dear Author,

Great work. I have borrowed your code to parse the coco json annotations using pandas data frame. Thank you for that.
I have some queries regarding your implementation. I see you have cropped the person and give it to training, the model only learns to detect single person in the image?

**Comment by robertklee** [comment]
Thanks, I'm glad to hear you found it helpful! Yes, you're right. It only does single person HPE. We were limited in available time as we only had 3 months and were enrolled in 4 other courses. Multi-person is a much more complex problem that we set as a stretch goal, but determined it would be too much to tackle.

**Comment by soans1994** [comment]
If i use same method and give the entire image for training does it work for multiple person?
For example,
Use the whole image 3 times with 3 different labels while training if there are 3 persons?

**Comment by robertklee** [comment]
Do you mean overlay the labels?

Then you'll need to output a variable number of joints depending on # of ppl, and also assign which joint belongs to which person's skeleton. 

It's doable but the models are much more complex. There are better approaches.

**Comment by soans1994** [comment]
i have read about part affinity techniques to group the keypoints after prediction (grouping techniques). (future plan)

But, i want to predict multiple keypoints heatmaps for multi-person in the first place, before applying post-processing. How can i do this?
Now, i am training and waiting for results, i hope the model learns to detect multi-person.

-my one training sample is as below.
- for first person in the image: give the entire image (with multi-person) to the model (resized to 256x256), but only 1 label corresponding the the first person.
- repeat procedure for all persons.

**Comment by robertklee** [comment]
i don't think that'll work. by the way, don't forget to credit our repo for code you used. :)

**Comment by soans1994** [comment]
thank you, I will credit your repo.

**Comment by soans1994** [comment]
please let me know if you find some information regarding multi-face deteciton.

thank you
